.github/workflows/Git-transport-tests.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+"""
+A script to:
+    1. Initialize a bare Git repository under ./repos/repo.git.
+    2. Enable HTTP receive-pack (push) for that repo.
+    3. Create an initial commit with "hello test1" in README.md and push to the bare repo.
+    4. Launch an HTTP Git server (using git-http-backend via CGI) on port 8000.
+    5. Clone the repo over HTTP, update README.md to "hello world", commit, and push back.
+    6. Verify the push by cloning via file:// and checking the README.md contents.
+
+purpose of script:
+    To show that building HTTP git server (that supports push/pull) is possible but adding "gittuff::" creates an infinite stall for uknown reason.
+"""
+import os
+import shutil
+import threading
+import subprocess
+import sys
+import time
+
+# Settings
+env = os.environ
+PORT = 8000
+BASE_DIR = os.getcwd()
+BARE_PARENT = os.path.join(BASE_DIR, 'repos')
+BARE_REPO = os.path.join(BARE_PARENT, 'repo.git')
+WORK_INITIAL = os.path.join(BASE_DIR, 'work1')
+WORK_HTTP = os.path.join(BASE_DIR, 'work2')
+WORK_VERIFY = os.path.join(BASE_DIR, 'verify')
+CGI_DIR = os.path.join(BASE_DIR, 'cgi-bin')
+
+# set a fallback identity so commits don’t fail in CI
+subprocess.check_call(["git", "config", "--global", "user.email", "action@github.com"])
+subprocess.check_call(["git", "config", "--global", "user.name",  "GitHub Action"])
+
+def run(cmd, cwd=None):
+    print(f"[*] Running: {' '.join(cmd)}")
+    subprocess.check_call(cmd, cwd=cwd)
+
+
+def run_http_server(repo_root, port):
+    # Prepare CGI directory and link git-http-backend
+    os.makedirs(CGI_DIR, exist_ok=True)
+    backend = shutil.which('git-http-backend') or shutil.which('git-http-backend.exe')
```

Comment:
Do we need this check? Isn't this part of the git binary? `git http-backend` may work?